plugins/plugin-docker/che-plugin-docker-client/src/test/java/org/eclipse/che/plugin/docker/client/DockerConnectorTest.java

```diff
@@ -107,4 +109,32 @@ public void shouldProduceAnErrorWhenPutsResourcesIntoContainer() throws IOExcept
 
         dockerConnector.putResource("id", "path", source, false);
     }
+
+    @Test
+    public void shouldParseDigestFromDockerPushOutput() throws IOException, InterruptedException {
+        String digest = "hash:1234567890";
+        String repository = "repository";
+        String tag = "tag";
+        String registry = "registry";
+
+        when(initialAuthConfig.getAuthConfigHeader()).thenReturn("authHeader");
+        when(dockerResponse.getStatus()).thenReturn(200);
+        when(dockerResponse.getInputStream()).thenReturn(new InputStream() {
+            private int i = 0;
+            private String stream = "{\"progress\":\"[=====>              ] 25%\"}\n" +
+                                    "{\"status\":\"Image already exists\"}\n" +
+                                    "{\"progress\":\"[===============>    ] 75%\"}\n" +
+                                    "{\"status\":\"" + tag + ": digest: " + digest + " size: 12345\"}";
+            @Override
+            public int read() throws IOException {
+                return (i < stream.length()) ? stream.charAt(i++) : -1;
+            }
+        });
+
+        Assert.assertEquals(digest, dockerConnector.push(repository,
```

Comment:
I strongly recommend you to use static import for `assertEquals` and other assert* methods or util methods at all.